jax/_src/numpy/lax_numpy.py

```diff
@@ -300,9 +300,11 @@ def _result_dtype(op, *args):
   return _dtype(op(*args))
 
 
-def _arraylike(x): return isinstance(x, ndarray) or isscalar(x)
+def _arraylike(x):
+  return isinstance(x, ndarray) or isscalar(x) or hasattr(x, '__jax_array__')
```

Comment:
Would it be worth checking that `__jax_array__` is callable?

Reply:
Hmm, interesting, but that seems a bit over-defensive; if it's not callable, then we'll almost immediately after this get a callability error from trying to call it.